plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineWtpProjectTest.java

```diff
@@ -143,22 +141,17 @@ private boolean containsSegment(IPath path, String segment) {
 
   @Test
   public void testNoTestClassesInDeploymentAssembly()
-      throws InvocationTargetException, CoreException {
+      throws InvocationTargetException, CoreException, InterruptedException {
     CreateAppEngineWtpProject creator = newCreateAppEngineWtpProject();
     creator.execute(monitor);
 
-    assertNoTestClassesInDeploymentAssembly();
-  }
-
-  private void assertNoTestClassesInDeploymentAssembly() {
-    StructureEdit core = StructureEdit.getStructureEditForRead(project);
-    try {
-      WorkbenchComponent component = core.getComponent();
-      assertEquals(0, component.findResourcesBySourcePath(new Path("/src/test/java"), 0).length);
-      assertEquals(1, component.findResourcesBySourcePath(new Path("/src/main/java"), 0).length);
-    } finally {
-      core.dispose();
+    while (!creator.deployAssemblyEntryRemoveJob.jobCompleted) {
+      Thread.sleep(50);
```

Comment:
It is guaranteed to terminate unless some jobs hangs with a deadlock (which shouldn't happen), but I see no harm in waiting for a set time. I'll make the job a system job so that it's not affected by `NonSystemJobSuspendeder`.